pkg/fanal/walker/walk_test.go

```diff
@@ -0,0 +1,120 @@
+package walker
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer/os"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_shouldSkipFile(t *testing.T) {
+	if runtime.GOOS == os.Windows {
+		t.Skip("glob paths are not supported on Windows")
+	}
```

Comment:
after some testing, I found out that globs aren't really supported on Windows natively. See here https://stackoverflow.com/a/72468013

@knqyf263 do you have any other ideas? My vote will be to just add a section in the readme that says this feature is only supported on Linux/MacOS.

Reply:
Hmm. I took a read on the source code, but it looks like Windows is supported.
https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/path/filepath/match.go;l=43

There is one note, though.

>On Windows, escaping is disabled. Instead, '\\' is treated as path separator.

https://pkg.go.dev/path/filepath#Match

How don't globs work? Do you have an example?

Reply:
You can see the failing test here on windows 

https://github.com/aquasecurity/trivy/actions/runs/4493776917/jobs/7905412196?pr=3866

Maybe we need to define the globs differently on windows? I will try to spin up a local VM to test this out.

Reply:
OK I've figured this out. Seems like using `path.Match` is better in this case as we already (internally within Trivy) process all paths with forward slash. Tests should be good now.

Here's the reference that I missed earlier https://pkg.go.dev/path/filepath:
> The filepath package uses either forward slashes or backslashes, depending on the operating system. To process paths such as URLs that always use forward slashes regardless of the operating system, see the path package.

Reply:
Sorry didn't see your new comment. 

So in short, this part of the code uses `filepath.ToSlash` in Trivy for all OS's. https://github.com/aquasecurity/trivy/blob/f1fb1da779231c41441154662b48923ac68bb5c0/pkg/fanal/walker/walk.go#L33-L52

This is perfectly fine, as it internally converts all paths use forward slashes regardless of the OS. I wasn't aware of this. So therefore in our case, we can simply use `path.Match` as that's the recommendation per the Go documentation.